### DISCUSSION TITLE
blkid fails with error 512

### DISCUSSION THREAD

**Discussion by parsley72** [util-linux/util-linux#3271] [Q&A]
Sometimes when I run `blkid --match-token TYPE=crypto_LUKS --output device /dev/mmcblk2p1` it fails with error 512. I can't find in the manual or source code what this means. Any idea?

```
# blkid --version                      
blkid from util-linux 2.37.4  (libblkid 2.37.4, 14-Feb-2022)
```

**Comment by karelzak** [comment]
The best is to use `strace` or `LIBBLKID_DEBUG=all` to get more details.

**Comment by parsley72** [comment]
Unfortunately this is from logs of embedded devices in the field so I don't have that option. Can someone point me to where in the code error 512 can be returned?
